node/node_test.go

```diff
@@ -244,6 +245,83 @@ func TestRegisterProtocols_OpenRPC_WS(t *testing.T) {
 	}
 }
 
+func discoverCompliesOpenRPCSpecTestRunner(specDefinition string) error {
+	stack, err := New(testNodeConfig())
+	if err != nil {
+		return fmt.Errorf("failed to create protocol stack: %v", err)
+	}
+	defer stack.Close()
+
+	datadir := filepath.Join(os.TempDir(), "node-test")
+	defer os.RemoveAll(datadir)
+
+	stack.config = &DefaultConfig
+	stack.config.HTTPHost = DefaultHTTPHost
+	// enable all available APIs
+	stack.config.HTTPModules = []string{"admin", "debug", "eth", "ethash", "miner", "net", "personal", "rpc", "trace", "txpool", "web3"}
+	stack.config.DataDir = datadir
+
+	if err := stack.Start(); err != nil {
+		return fmt.Errorf("failed to start protocol stack: %v", err)
+	}
+	defer stack.Close()
+
+	client, err := rpc.Dial("http://localhost:8545")
+	if err != nil {
+		return fmt.Errorf("failed to create RPC client: %v", err)
+	}
+	defer client.Close()
+
+	res := make(map[string]interface{})
+	err = client.Call(&res, "rpc_discover")
+	if err != nil {
+		return fmt.Errorf("client call: %v", err)
+	}
+
+	schemaLoader := gojsonschema.NewSchemaLoader()
+	schemaLoader.Draft = gojsonschema.Draft7 // force schema version
+	schemaLoader.AutoDetect = false
+
+	schema, err := schemaLoader.Compile(gojsonschema.NewReferenceLoader(specDefinition))
+
+	documentLoader := gojsonschema.NewGoLoader(res)
+
+	result, err := schema.Validate(documentLoader)
+	if err != nil {
+		return fmt.Errorf("failed to validate OpenRPC schema: %v", err)
+	}
+
+	if !result.Valid() {
+		for _, desc := range result.Errors() {
+			fmt.Printf("- %s\n", desc)
+		}
+		return fmt.Errorf("the OpenRPC discovery document is not compliant with spec")
+	}
+	return nil
+}
+
+// TestOpenRPCMetaSchemaSpec tests compliance of discovery endpoint with the OpenRPC spec stored locally
+func TestOpenRPCMetaSchemaSpec(t *testing.T) {
+	openRPCSpecFile := "file://./testdata/open-rpc-meta-schema-1.14.0.json"
+	if err := discoverCompliesOpenRPCSpecTestRunner(openRPCSpecFile); err != nil {
+		t.Fatal(err)
+	}
+}
+
+var openRPCSpecLatestTestEnvVarName = "COREGETH_OPENRPC_LATEST_SPEC_ENDPOINT"
+
+// TestOpenRPCMetaSchemaSpecLatest tests compliance of discovery endpoint with the OpenRPC latest spec loaded remotelly as set in the CI
+func TestOpenRPCMetaSchemaSpecLatest(t *testing.T) {
```

Comment:
- [ ] `COREGETH_OPENRPC_LATEST_SPEC_ENDPOINT` <- this still needs to be installed at the CI environment, or?
- [ ] should this test check either or both?
  + equivalence of remote and local schemas
  + equivalence of remote and local version numbers

Reply:
Yes it has to be added at the CI, I had this in mind for after the merge.

Hmm, nice idea to also have something that checks if the local testdata file needs to be updated. Though at this point, that there is no endpoint to point on the latest/master JSON file, we are in control of both endpoints, which will probably point at the same location.

Now that I have written the above, I thought that I understood your intention wrongly. Do you want this test, to test using an external endpoint, or to compare the external latest JSON with the local one?

Reply:
:) Originally I had the 
> compare the external latest JSON with the local one

in mind (since theoretically it should be exactly equivalent as using the local copy given the pegged version), but if there's a way to use the _latest_ remote version (instead of always replicating the local copy) then that seems like it would be valuable too.

Reply:
Solved by https://github.com/etclabscore/core-geth/pull/295/commits/386e26eb9eb96d713b9e43bb2474f11f32c8edf2

@meowsbits I added a test that compares the local file with the latest **released** one.
I removed the test that checks for discovery compliance with remote spec, as I prefer more that we are aware of the failing test (changed spec) and we fix it, instead of ignoring the local test, because the remote one passes.

```shell
COREGETH_OPENRPC_LATEST_SPEC_ENDPOINT=https://github.com/open-rpc/meta-schema/releases/latest/download/open-rpc-meta-schema.json CI=true go test -run ^TestOpenRPCMetaSchemaSpecLatest$ github.com/ethereum/go-ethereum/node
```

Reply:
Yeap, nice. LGTM.